test/opt/test_set_spec_const_default_value.cpp

```diff
@@ -0,0 +1,165 @@
+          "%1 = OpSpecConstant %int 2147483647\n",
+        },
+        // 4. 64-bit uint type
+        {
+          // code
+          "OpDecorate %1 SpecId 100\n"
+          "%ulong = OpTypeInt 64 0\n"
+          "%1 = OpSpecConstant %ulong 100\n",
+          // default values
+          SpecIdToValueMap{{100, {0xffffffff, 0xffffffff}}},
+          // expected
+          "OpDecorate %1 SpecId 100\n"
+          "%ulong = OpTypeInt 64 0\n"
+          "%1 = OpSpecConstant %ulong 18446744073709551615\n",
+        },
+        // 5. 32-bit float type
+        {
+          // code
+          "OpDecorate %1 SpecId 101\n"
+          "%float = OpTypeFloat 32\n"
+          "%1 = OpSpecConstant %float 200\n",
+          // default values
+          SpecIdToValueMap{{101, {0xffffffff}}},
+          // expected
+          "OpDecorate %1 SpecId 101\n"
+          "%float = OpTypeFloat 32\n"
+          "%1 = OpSpecConstant %float -0x1.fffffep+128\n",
+        },
+        // 6. 64-bit float type
+        {
+          // code
+          "OpDecorate %1 SpecId 201\n"
+          "%double = OpTypeFloat 64\n"
+          "%1 = OpSpecConstant %double 3.14159265358979\n",
+          // default values
+          SpecIdToValueMap{{201, {0xffffffff, 0x7fffffff}}},
+          // expected
+          "OpDecorate %1 SpecId 201\n"
+          "%double = OpTypeFloat 64\n"
+          "%1 = OpSpecConstant %double 0x1.fffffffffffffp+1024\n",
+        },
+        // 7. SpecId not found, expect no modification.
+        {
+          // code
+          "OpDecorate %1 SpecId 201\n"
+          "%double = OpTypeFloat 64\n"
+          "%1 = OpSpecConstant %double 3.14159265358979\n",
+          // default values
+          SpecIdToValueMap{{8888, {0xffffffff, 0x7fffffff}}},
+          // expected
+          "OpDecorate %1 SpecId 201\n"
+          "%double = OpTypeFloat 64\n"
+          "%1 = OpSpecConstant %double 3.14159265358979\n",
+        },
+        // 8. Multiple spec constants.
+        {
+          // code
+          "OpDecorate %1 SpecId 201\n"
+          "OpDecorate %2 SpecId 202\n"
+          "OpDecorate %3 SpecId 203\n"
+          "%bool = OpTypeBool\n"
+          "%int = OpTypeInt 32 1\n"
+          "%double = OpTypeFloat 64\n"
+          "%1 = OpSpecConstant %double 3.14159265358979\n"
+          "%2 = OpSpecConstant %int 1024\n"
+          "%3 = OpSpecConstantTrue %bool\n",
+          // default values
+          SpecIdToValueMap{
+            {201, {0xffffffff, 0x7fffffff}},
+            {202, {2048}},
+            {203, {false}},
+          },
+          // expected
+          "OpDecorate %1 SpecId 201\n"
+          "OpDecorate %2 SpecId 202\n"
+          "OpDecorate %3 SpecId 203\n"
+          "%bool = OpTypeBool\n"
+          "%int = OpTypeInt 32 1\n"
+          "%double = OpTypeFloat 64\n"
+          "%1 = OpSpecConstant %double 0x1.fffffffffffffp+1024\n"
+          "%2 = OpSpecConstant %int 2048\n"
+          "%3 = OpSpecConstantFalse %bool\n",
+        },
+    }));
```

Comment:
Add tests to show that true is returned when the pass modifies, and false when it does not modify.